### ISSUE TITLE
Mention 2015 edition in "note: `?` is not a macro repetition operator" errors

### ISSUE THREAD

**Issue by SimonSapin** [rust-lang/rust#58050]
Example error message:

```rust
error: expected `*` or `+`
   --> src/libcore/macros.rs:767:41
    |
767 |             impl $( <$( $lifetime ),+> )? Fn<(&'a str,)> for $Name {
    |                                         ^
    |
    = note: `?` is not a macro repetition operator
```

But it is! I couldn’t figure it out without asking on IRC what’s going on. It is a repetition operator… but not in the 2015 edition. Switching to 2018 would make this error go away. 

The error message should mention this.

**Comment by estebank** [comment]
*Mentoring notes:*

There are three notes in https://github.com/rust-lang/rust/blob/master/src/libsyntax/ext/tt/quoted.rs that need to be updated to read

```
`?` is not a macro repetition operator in the 2015 edition, but accepted in the 2018 edition
```

After doing the code change you need to run `./x.py test src/test/ui --stage 1 --bless` to update the `stderr` output files.

**Comment by polybuildr** [comment]
It looks like this was fixed in PR #58075 and merged by bors. Should this issue be closed now?
